dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,132 @@
+  Here are the affects of each property on the IOMUXC SW_PAD_CTL register:
+
+  input-schmitt-enable: HYS=1
+  drive-open-drain: ODE=1
+  input-enable: SION=1 (in SW_MUX_CTL_PAD register)
+  bias-bus-hold: PKE=1, PUE=0
+  bias-pull-down: PKE=1, PUE=1, PUS=0
+  bias-pull-up: PKE=1, PUE=1, PUS=<prop_value>
+  slew-rate: SRE=<prop_value>
+  drive-strength: DSE=<prop_value>
+  nxp,speed: SPEED=<prop_value>
+
+
+
+compatible: "nxp,mcux-rt-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - input-schmitt-enable
+            - drive-open-drain
+            - input-enable
+            - bias-bus-hold
+            - bias-pull-down
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+  description: MCUX RT pin controller pin group
+  child-binding:
+    description: |
+      MCUX RT pin controller pin configuration node.
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of values defining the configuration register for a given
+          pinmux mode. See fsl_iomuxc.h in the NXP HAL for a list.
+          The array takes the following format:
+          mux_register, mux_value, input_register, input_daisy, config_register
+      drive-strength:
+        required: false
+        type: int
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+          - 4
+          - 5
+          - 6
+          - 7
+        description: |
+          Pin output drive strength. Sets the DSE field in the IOMUXC peripheral.
+          000 DSE_0_output_driver_disabled_ — output driver disabled
+          001 DSE_1_R0_1 — 157 Ohm impedance @3.3V, 260 Ohm impedance @1.8V
+          010 DSE_2_R0_2 — 78 Ohm @3.3V, 130 Ohm @1.8V
+          011 DSE_3_R0_3 — 53 Ohm @3.3V, 88 Ohm @1.8V
+          100 DSE_4_R0_4 — 39 Ohm @3.3V, 65 Ohm @1.8V
+          101 DSE_5_R0_5 — 32 Ohm @3.3V, 52 Ohm @1.8V
+          110 DSE_6_R0_6 — 32 Ohm @3.3V, 43 Ohm @1.8V
+          111 DSE_7_R0_7 — 26 Ohm @3.3V, 37 Ohm @1.8V
+      bias-pull-up:
+        required: false
+        type: int
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+        description: |
+          Select the value of the pull up resistor present on this pin
+          Corresponds to the PUS field in the IOMUXC peripheral
+          00 Unused- no change will be applied to pin
+          01 PUS_1_47K_Ohm_Pull_Up — 47K Ohm Pull Up
+          10 PUS_2_100K_Ohm_Pull_Up — 100K Ohm Pull Up
+          11 PUS_3_22K_Ohm_Pull_Up — 22K Ohm Pull Up
+      slew-rate:
+        required: false
+        type: int
+        enum:
+          - 0
+          - 1
+        description: |
+          Select slew rate for pin. Corresponds to SRE field in IOMUXC peripheral
+          0 SRE_0_Slow_Slew_Rate — Slow Slew Rate
+          1 SRE_1_Fast_Slew_Rate — Fast Slew Rate
+      nxp,speed:
+        required: false
```

Comment:
I'm not sure what you mean here. I left all pin properties as `required: false` because the pinctrl driver uses `DT_PROP_OR` to default all these values to a special "none" value if they are not set explicitly in the devicetree. This allows the driver to skip configuring any bit fields in the pin configuration register that the user does not explicitly set with a devicetree property.

So if `nxp,speed` is set in the devicetree, the driver will pick that up and write the provided value to the pin configuration register. Otherwise it will leave the register at its default value.